src/main.rs

```diff
@@ -49,7 +49,15 @@ fn main() -> io::Result<()> {
 
     let bin_name = prog.value_of("binary").unwrap();
 
-    let scale: usize = prog.value_of("scale").unwrap().parse().unwrap();
+    let scale: usize = match prog.value_of("scale")
+                            .unwrap()
+                            .parse() {
+                                Ok(size) => size,
```

Comment:
Please handle the exception of 0, also I think we should give a up-limit like 4.